sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -123,9 +123,12 @@ abstract class OffsetWindowFunctionFrameBase(
   protected val fillDefaultValue = {
     // Collect the expressions and bind them.
     val boundExpressions = Seq.fill(ordinal)(NoOp) ++ expressions.toImmutableArraySeq.map { e =>
-      if (e.default == null || e.default.foldable && e.default.eval() == null) {
+      if (e.default == null) {
         // The default value is null.
         Literal.create(null, e.dataType)
+      } else if (e.default.foldable) {
```

Comment:
The change looks reasonable to me, just to double check: ideally the optimizer should have already optimized foldable expressions into literals, so this change won't have any real impact?

Reply:
Yes. The optimizer has already optimized the foldable expressions into literals. It means the default expression of Lead/Lag is a literal after optimization. But the Lead/Lag is still there.

Reply:
After a second thought, we can simplify these code as follows:
```
      if (e.default == null) {
        // The default value is null.
        Literal.create(null, e.dataType)
      } else {
        // The default value is an expression.
        BindReferences.bindReference(e.default, inputAttrs)
      }
```
But the current change looks more safe.